src/vulkan/vk-heap.cpp

```
@@ -55,22 +61,35 @@ HeapImpl::PageImpl::PageImpl(Heap* heap, const PageDesc& desc, DeviceImpl* devic
+    // alignment chosen by the Vulkan driver for the backing buffer.
+    Size bufferSize = m_desc.size + m_desc.alignment - 1;
+    SLANG_RETURN_ON_FAIL(m_buffer.init(api, bufferSize, usage, reqMemoryProperties, externalMemoryHandleTypeFlags));
+
+    VkBufferDeviceAddressInfo info = {};
+    info.sType = VK_STRUCTURE_TYPE_BUFFER_DEVICE_ADDRESS_INFO;
+    info.buffer = m_buffer.m_buffer;
+    DeviceAddress bufferAddress = (DeviceAddress)api.vkGetBufferDeviceAddress(api.m_device, &info);
+    if (bufferAddress == 0)
+    {
+        return SLANG_FAIL;
+    }
```

Cppcheck (2.21.0)

[style] The function 'getDumpDir' is never used.

(unusedFunction)